src/tools/html-checker/main.rs

```diff
@@ -0,0 +1,96 @@
+use std::env;
+use std::path::Path;
+use std::process::{Command, Output};
+
+fn check_html_file(file: &Path) -> usize {
+    let to_mute = &[
+        // "disabled" on <link> or "autocomplete" on <select> emit this warning
+        "PROPRIETARY_ATTRIBUTE",
+        // It complains when multiple in the same page link to the same anchor for some reason...
+        "ANCHOR_NOT_UNIQUE",
+        // If a <span> contains only HTML elements and no text, it complains about it.
+        "TRIM_EMPTY_ELEMENT",
+        // FIXME: the three next warnings are about <pre> elements which are not supposed to
+        //        contain HTML. The solution here would be to replace them with a <div> with
+        //        ""
+        "MISSING_ENDTAG_BEFORE",
+        "INSERTING_TAG",
+        "DISCARDING_UNEXPECTED",
+        // FIXME: mdbook repeats the name attribute on <input>. When the fix is merged upstream,
+        //        this warning can be used again.
+        "REPEATED_ATTRIBUTE",
+        // FIXME: mdbook uses "align" attribute on <td>, which is not allowed.
+        "MISMATCHED_ATTRIBUTE_WARN",
+        // FIXME: mdbook doesn't add "alt" attribute on images.
+        "MISSING_ATTRIBUTE",
+        // FIXME: mdbook doesn't escape `&` (in "&String" for example).
+        "UNKNOWN_ENTITY",
+        // Compiler docs have some inlined <style> in the markdown.
```

Comment:
Could these be moved to --html-in-header or something like that? Or is the styling only for specific pages?

Reply:
It's in the markdown directly. So not really.

Reply:
It was in one of the books actually, so I removed it since we don't check them anymore.